include/kframework/foundry.md

```diff
@@ -158,6 +158,12 @@ The configuration of the Foundry Cheat Codes is defined as follwing:
     - `<expectedRevert>` flags if the next call is expected to revert or not.
     - `<expectedDepth>` records the depth at which the call is expected to revert.
     - `<expectedBytes>` keeps the expected revert message as a ByteArray.
+3. The `<expectOpcode>` subconfiguration stores values used for `expect*OPCODE*` cheat codes.
+    - `<isOpcodeExpected>` flags if a call opcode is expected.
+    - `<expectedAddress>` keeps the expected caller.
+    - `<expectedValue>` keeps expected `msg.value`.
+    - `<expectedData>` keeps expected `calldata`.
+    - `<opcodeType>` keeps track of what `CALL*` Opcode is expected.
 
 ```k
```

Comment:
I think we can probably implement `expectRevert` in terms of this more general machinery, and then remove the `expectRevert` subconfiguration and any other special handling related to that.

I would say if you don't want to implement it in this PR, make an issue for it.

Reply:
That would also enable us to remove the third argument to `foundry_success` (where `EXPECTEDREVERT_FINAL` is stored). But it would mean that you can't do an `expectRevert` and an `expectCall` at the same time. Maybe that is good though, if you can only do one `expect*` at a time?

Reply:
In the case of `expectRevert`, they specifically mention that `After calling expectRevert, calls to other cheat codes before the reverting call are ignored.`. This is not mentioned for `expectCall` or `expectEmit` cheat codes. My plan was to first merge this PR and the one implementing `expectEmit` using a similar subconfiguration. Next I'm going to open a new PR to refactor the `expect*` subconfigurations.